### PR TITLE
Add ResultExt::map_err_to_string and others to preserve error message context

### PR DESCRIPTION
**Pull request by antifuchs**
This PR adds the following methods:

* `ResultExt::map_err_to_string` - to perform the (very common) conversion of Err(something that's Display) -> Err(error message as String), preserving all the context & additional information that we add with anyhow.
* `ResultExt::err_to_string` - to extract the error message from a Result reference (not sure this is useful, but it was possible)
* `DisplayExt::to_string_alt` - like to_string, but the equivalent of `format!("{:#}", obj)` instead of `{}`.

They all allow formatting error values as detailed strings, preserving any context that might have been added to the error.

Then, we adjust the call sites for `.map_err(|e| e.to_string())` to use `.map_err_to_string()` and do some more format-string readjustment to preserve as much context on errors as possible.

### CODE REVIEW

**Comment by antifuchs** [comment]
I guess early results indicate that this ain't it (yet), chief. I'll noodle on this some more.

**Comment by antifuchs** [comment]
(Heh, forgot to fix up imports on sqllogictest files. I'll fix those tomorrow)

File: src/testdrive/src/format/avro.rs [hunk 1]
```diff
@@ -220,10 +217,471 @@ pub fn from_json(json: &JsonValue, schema: SchemaNode) -> Result<Value, String>
...
+            },
+        ) => todo!(),
+        (
+            JsonValue::Object(_),
+            SchemaPiece::ResolveEnum {
+                doc,
+                symbols,
+                default,
+            },
+        ) => todo!(),
```
Comment (benesch) [inline src/testdrive/src/format/avro.rs:684]: What happened here? 🤔

**Review by benesch — COMMENTED** [review]
Awesome! LGTM assuming those lines in src/testdrive/src/format/avro.rs are a mistake?

File: src/testdrive/src/format/avro.rs (hunk 1 above)
Comment (antifuchs) [reply src/testdrive/src/format/avro.rs:684]: whaaaaaat. I imagine this was some rust-analyzer magic gone wild, I'll undo.

**Pull request merged**